addons/l10n_multilang/models/l10n_multilang.py

```diff
@@ -151,26 +151,29 @@ class BaseLanguageInstall(models.TransientModel):
 
     def lang_install(self):
         self.ensure_one()
-        already_installed = self.lang in [code for code, _ in self.env['res.lang'].get_installed()]
         res = super(BaseLanguageInstall, self).lang_install()
-        if already_installed:
+        lang_codes = set(self.lang_ids.mapped('code'))
+        installed = {code for code, __ in self.env['res.lang'].get_installed()}
+        to_install = lang_codes - installed
+        if not to_install:
             # update of translations instead of new installation
             # skip to avoid duplicating the translations
             return res
 
         # CoA in multilang mode
         for coa in self.env['account.chart.template'].search([('spoken_languages', '!=', False)]):
-            if self.lang in coa.spoken_languages.split(';'):
-                # companies on which it is installed
-                for company in self.env['res.company'].search([('chart_template_id', '=', coa.id)]):
-                    # write account.account translations in the real COA
-                    coa._process_accounts_translations(company.id, [self.lang], 'name')
-                    # write account.group translations
-                    coa._process_account_group_translations(company.id, [self.lang], 'name')
-                    # copy account.tax name translations
-                    coa._process_taxes_translations(company.id, [self.lang], 'name')
-                    # copy account.tax description translations
-                    coa._process_taxes_translations(company.id, [self.lang], 'description')
-                    # copy account.fiscal.position translations
-                    coa._process_fiscal_pos_translations(company.id, [self.lang], 'name')
+            coa_langs_codes = to_install & set(coa.spoken_languages.split(';'))
+            # companies on which it is installed
+            for company in self.env['res.company'].search([('chart_template_id', '=', coa.id)]):
```

Comment:
Shouldn't you do ``if coa_langs_codes`` before looping on companies ? No need to update companies if they don't use those languages (was done in previous code by the way)